src/main/java/org/spongepowered/common/data/AbstractArchetype.java

```diff
@@ -192,12 +199,24 @@ public DataTransactionResult copyFrom(DataHolder that, MergeFunction function) {
                 .flatMap(processor -> processor.readValue(this.data));
     }
 
+    @SuppressWarnings("unchecked")
+    @Override
+    public <R> Optional<R> getDefault(Key<? extends BaseValue<R>> key) {
+        return DataUtil.getNbtProcessor(this.getDataType(), key)
+                .flatMap(processor -> processor.readFrom(this.data).map(v -> v.getDefault()));
```

Comment:
multi-line the nested mapping. In the event there's an exception, we want to be able to pinpoint which lambda is having an issue.